appserver/web/weld-integration/src/main/java/org/glassfish/weld/DeploymentImpl.java

```diff
@@ -449,6 +449,10 @@ public BeanDeploymentArchive loadBeanDeploymentArchive(Class<?> beanClass) {
                            new Object[]{beanClass, newBda});
             }
             beanDeploymentArchives.add(newBda);
+
+            // Make the extension BDA visible to all previously found extensions
```

Comment:
It is other way around:  Make all previously found extensions visible to this newly created extension BDA